presto-tests/src/test/java/io/prestosql/connector/system/TestSystemConnector.java

```diff
@@ -100,6 +105,52 @@ public void tearDown()
         executor.shutdownNow();
     }
 
+    @Test
+    public void testRuntimeQueriesTimestamps()
+    {
+        ZonedDateTime timeBefore = ZonedDateTime.now();
+        computeActual("SELECT 1");
+        MaterializedResult result = computeActual("" +
+                "SELECT max(created), max(started), max(last_heartbeat), max(\"end\") " +
+                "FROM system.runtime.queries");
+        ZonedDateTime timeAfter = ZonedDateTime.now();
+
+        MaterializedRow row = Iterables.getOnlyElement(result.toTestTypes().getMaterializedRows());
+        List<Object> fields = row.getFields();
+        assertThat(fields).hasSize(4);
+        for (int i = 0; i < fields.size(); i++) {
+            Object value = fields.get(i);
+            assertThat((ZonedDateTime) value)
+                    .as("value for field " + i)
+                    .isNotNull()
+                    .isAfterOrEqualTo(timeBefore)
+                    .isBeforeOrEqualTo(timeAfter);
+        }
+    }
+
+    @Test
+    public void testRuntimeTasksTimestamps()
+    {
+        ZonedDateTime timeBefore = ZonedDateTime.now();
+        computeActual("SELECT 1");
+        MaterializedResult result = computeActual("" +
+                "SELECT max(created), max(start), max(last_heartbeat), max(\"end\") " +
+                "FROM system.runtime.tasks");
+        ZonedDateTime timeAfter = ZonedDateTime.now();
+
+        MaterializedRow row = Iterables.getOnlyElement(result.toTestTypes().getMaterializedRows());
+        List<Object> fields = row.getFields();
+        assertThat(fields).hasSize(4);
+        for (int i = 0; i < fields.size(); i++) {
+            Object value = fields.get(i);
+            assertThat((ZonedDateTime) value)
+                    .as("value for field " + i)
+                    .isNotNull()
+                    .isAfterOrEqualTo(timeBefore)
+                    .isBeforeOrEqualTo(timeAfter);
+        }
+    }
+
```

Comment:
can you add test for `system.runtime.transactions`?

Reply:
added